src/vcpkg/tools.cpp

```diff
@@ -726,171 +598,291 @@ coscli version v0.11.0-beta
+                    // We use cmake as the core extractor on Windows, so we need to perform a special dance when
+                    // extracting it.
+                    win32_extract_bootstrap_zip(fs, *this, download_path, tool_dir_path);
                 }
-
-                if (tool == Tools::IFW_REPOGEN)
+                else
+#endif // ^^^ _WIN32
                 {
-                    auto installer_base = get_tool_path(paths, Tools::IFW_INSTALLER_BASE);
-                    installer_base.replace_filename("repogen.exe");
-                    return installer_base;
+                    extract_archive(fs, *this, download_path, tool_dir_path);
                 }
+            }
+            else
+            {
+                fs.create_directories(exe_path.parent_path(), IgnoreErrors{});
+                fs.rename(download_path, exe_path, IgnoreErrors{});
+            }
 
-                return get_tool_pathversion(paths, tool).p;
-            });
+            Checks::check_exit(
+                VCPKG_LINE_INFO, fs.exists(exe_path, IgnoreErrors{}), "Expected %s to exist after fetching", exe_path);
+
+            return exe_path;
         }
 
-        const PathAndVersion& get_tool_pathversion(const VcpkgPaths& paths, StringView tool) const
+        const std::string& get_config_contents() const
         {
-            return path_version_cache.get_lazy(tool, [&]() -> PathAndVersion {
-                // First deal with specially handled tools.
-                // For these we may look in locations like Program Files, the PATH etc as well as the auto-downloaded
-                // location.
-                if (tool == Tools::CMAKE)
-                {
-                    if (get_environment_variable("VCPKG_FORCE_SYSTEM_BINARIES").has_value())
-                    {
-                        return {"cmake", "0"};
-                    }
-                    return get_path(paths, CMakeProvider(), abiToolVersionHandling == RequireExactVersions::YES);
-                }
-                if (tool == Tools::GIT)
-                {
-                    if (get_environment_variable("VCPKG_FORCE_SYSTEM_BINARIES").has_value())
-                    {
-                        return {"git", "0"};
-                    }
-                    return get_path(paths, GitProvider());
-                }
-                if (tool == Tools::NINJA)
-                {
-                    if (get_environment_variable("VCPKG_FORCE_SYSTEM_BINARIES").has_value())
-                    {
-                        return {"ninja", "0"};
-                    }
-                    return get_path(paths, NinjaProvider());
-                }
-                if (tool == Tools::POWERSHELL_CORE)
+            return xml_config_contents.get_lazy(
+                [this]() { return this->fs.read_contents(this->xml_config, VCPKG_LINE_INFO); });
+        }
+
+        virtual const Path& get_tool_path(StringView tool) const override { return get_tool_pathversion(tool).p; }
```

Comment:
I like this overload because 90% of callers just want the path so they can proceed to build a command out of it.

If you would like to see `get_tool_version()` replaced with `get_tool_pathversion().version` while keeping `get_tool_path()`, I can do that.

Reply:
You could make it nonvirtual and in the interface if it's to make callers prettier.

Reply:
That would be fine by me -- I thought you hated having _any_ code in the interface though?